crates/oci/src/client.rs

```diff
@@ -230,10 +300,51 @@ impl Client {
                             .await
                         {
                             Err(e) => return Err(e),
+                            // TODO: handle error(s), avoid unwraps below
```

Comment:
Hmm yeah there is all kinds of weirdness going on in here (already)...

Reply:
I won't claim it isn't still weird, but at least I got rid of the `unwrap()`s by breaking it out into a separate fn in https://github.com/fermyon/spin/pull/1847/commits/71640b3837e5da7dd776f2aecf263188a82a0934